nixos/modules/security/acme/default.nix

```diff
@@ -545,7 +545,7 @@ let
       };
 
       server = mkOption {
-        type = types.str;
+        type = types.nullOr types.str;
```

Comment:
I think we should add a warning to nudge people to set this explicitly to safeguard us against future changes in lego.

Reply:
By the same logic, I doubt that lego would ever make such a change due to the implications for their end users. I also think that users who now set the value to null understand that they adopt the default from Lego and won't need a warning to remind them.

Reply:
When we're talking about catching future changes, we should probably also add a test case for `null` to https://github.com/NixOS/nixpkgs/pull/286999 to catch if someone decides to drop this special case *again*.